src/main/java/seedu/duke/operationlist/FlightList.java

```diff
@@ -0,0 +1,344 @@
+package seedu.duke.operationlist;
+
+import seedu.duke.terminalinfo.FlightInfo;
+import seedu.duke.exceptions.SkyControlException;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class FlightList extends OperationList {
+    public static int flightIndex = 0;
+    private static final String FLIGHT_ADD_COMMAND = "flight add";
+    private static final String FLIGHT_ADD_DELIMITER = "flight add ";
+    private static final String FLIGHT_DELETE_COMMAND = "flight delete";
+    private static final String FLIGHT_NUMBER_DELIMITER = " fn/";
+    private static final String AIRLINE_DELIMITER = " a/";
+    private static final String DESTINATION_DELIMITER = " d/";
+    private static final String DEPARTURE_TIME_DELIMITER = " dt/";
+    private static final String GATE_NUMBER_DELIMITER = " gn/";
+    private static final String CHECK_IN_ROW_DELIMITER = " c/";
+
+    protected static final String EMPTY_STRING = "";
+    protected static final int CHECK_MISSING_DETAIL = 0;
+    protected static final int AIRLINE_LENGTH_LIMIT = 22;
+    protected static final int DESTINATION_LENGTH_LIMIT = 22;
+    protected static boolean isFlightNumberPresent = false;
+    protected static boolean isDepartureTimePresent = false;
+    protected static boolean isGateNumberPresent = false;
+    protected static String timeRegex = "([01]?[0-9]|2[0-3])[0-5][0-9]";
+    protected static String checkInRegex = "[0-9]{2}-[0-9]{2}";
+    protected static String gateNumRegex = "^(\\d?[0-9]|[1-9]0)$";
+    protected String flightNumber;
+    protected String airline;
+    protected String destination;
+    protected String departureTime;
+    protected String gateNumber;
+    protected String checkIn;
+    protected String oldDepartureTime;
+
+    protected static int numOfFlights = 0;
+
+    //@@author Franky4566
+    public static void checkCommandLength(String description) throws SkyControlException {
+        if (description.isEmpty()) {
+            throw new SkyControlException(ui.showEmptyDescriptionMessage());
+        }
+    }
+
+    public static String extractDetail(String command, String start, String end) throws SkyControlException {
+
+        String extractedDetail;
+        int startIndex = command.indexOf(start) + start.length();
+        int endIndex = command.lastIndexOf(end);
+        checkNoDetailsMissing(startIndex, endIndex);
+        extractedDetail = command.substring(startIndex, endIndex).trim();
+        if (extractedDetail.equals(EMPTY_STRING)) {
+            throw new SkyControlException(ui.getMissingDetailsError());
+        }
+        return extractedDetail;
+    }
+
+    public static void checkNoDetailsMissing(int startIndex, int endIndex) throws SkyControlException {
+        if (endIndex <= startIndex || endIndex < CHECK_MISSING_DETAIL || startIndex < CHECK_MISSING_DETAIL) {
```

Comment:
Avoid complicateed expressions. Consider putting the checks into a seperate method.